doc/examples/setup_assistant/setup_assistant_tutorial.rst

```diff
@@ -232,97 +280,169 @@ some special operations to happen on them internally.
 
 * Leave *Parent Group* blank.
 
-.. image:: setup_assistant_panda_end_effector_add.png
+.. image:: setup_assistant_panda_add_end_effector.png
    :width: 700px
 
 Step 7: Add Passive Joints
 --------------------------
 
-The passive joints tab is meant to allow specification of any passive
-joints that might exist in a robot. These are joints that are unactuated
-on a robot (e.g. passive casters.) This tells the planners that they
-cannot (kinematically) plan for these joints because they can't be
-directly controlled. The Panda does not have any passive
-joints so we will skip this step.
+The passive joints pane is meant to allow specification of any passive
+joints that might exist in a robot. These are joints that are unactuated, 
+meaning that they cannot be directly controlled. It's important to specify
+passive joints so that the planners are aware of their existence and can avoid
+planning for them. If the planners do not know about the passive joints, they 
+might try to plan trajectories that involve moving the passive joints, which would
+result in invalid plans. The Panda robot arm does not have any passive joints so we will skip this step.
 
+Step 8: ros2_control URDF Modification
+--------------------------------------
 
-Step 8: 3D Perception
---------------------------
+The ros2_control URDF Modification helps modify the robot URDF to work with 
+`ros2_control <https://control.ros.org/master/index.html>`_.
 
-The 3D Perception tab is meant to set the parameters of the YAML configuration file
-for configuring the 3D sensors **sensors_3d.yaml**.
+This modification adds tags for command and state interfaces for each joint in the defined move groups. 
+The command_interface tags define the types of commands that can be sent to control the joint.
+The state_interface tags define the types of state information that can be read from the joint.
 
-e.g. **point_cloud** parameters:
+By default the Setup Assistant assumes **position** command interface 
+and **position** and **velocity** state interfaces, and we will proceed with this setting. 
 
-.. image:: setup_assistant_panda_3d_perception_point_cloud.png
+.. image:: ros2_control/setup_assistant_ros2_control_tags.png
    :width: 700px
 
-For more details about those parameters please see perception pipeline :doc:`tutorial </doc/examples/perception_pipeline/perception_pipeline_tutorial>`
+If necessary, select the desired command or state interfaces for your robot joints and 
+then click the **Add Interface** button.
+
+Step 9: ROS 2 Controllers
+-------------------------
 
-In case of **sensors_3d.yaml** was not needed, choose **None**.
+The ROS 2 control is a framework for (real-time) control of robots using. 
```

Comment:
```suggestion
ROS 2 control is a framework for (real-time) control of robots using. 
```

also this sentence ends abruptly. Using what?

Reply:
Sorry, this was a mistake. I rewrote it.